src/cli/cli.cpp

```diff
@@ -1887,14 +1887,29 @@ otError Interpreter::ProcessLeaderPartitionId(uint8_t aArgsLength, char *aArgs[]
 
     if (aArgsLength == 0)
     {
-        OutputLine("%u", otThreadGetLocalLeaderPartitionId(mInstance));
+        OutputLine("%u", otThreadGetLeaderPartitionId(mInstance));
     }
-    else
+    else if (strcmp(aArgs[0], "preferred") == 0)
     {
-        uint32_t partitionId;
+        if (aArgsLength == 1)
+        {
+            OutputLine("%u", otThreadGetPreferredLeaderPartitionId(mInstance));
+        }
+        else if (aArgsLength == 2)
+        {
+            uint32_t partitionId;
 
-        SuccessOrExit(error = ParseAsUint32(aArgs[0], partitionId));
-        otThreadSetLocalLeaderPartitionId(mInstance, partitionId);
+            SuccessOrExit(error = ParseAsUint32(aArgs[1], partitionId));
+            otThreadSetPreferredLeaderPartitionId(mInstance, partitionId);
+        }
+        else
+        {
+            error = OT_ERROR_INVALID_COMMAND;
```

Comment:
nit: Set error to `OT_ERROR_INVALID_COMMAND` by default?

Reply:
Updated